src/state/StateNode.java

```diff
@@ -96,6 +97,7 @@ public String getDiffString(StateNode other) {
         SortedMap<String, List<String>> otherState = other.state;
 
         StringBuilder sb = new StringBuilder();
+        sb.append(String.format("\nS%d\n----", id));
```

Comment:
It's slightly inaccurate to print just one id for the diff string, since the contents do not represent the entire state.

I think it might be better to print `{prevID} -> {currID}` for more accuracy.